### PR TITLE
Adding additional restriction types to app authentication method policy

### PR DESCRIPTION
**Pull request by madansr7**

### CODE REVIEW

File: api-reference/beta/api/appManagementPolicy-post.md [hunk 1]
```diff
@@ -55,7 +55,8 @@ If successful, this method returns a `201 Created` response code with the new [a
 The following is an example of the request. This request created an app management policy with the following settings:
 
 - Enables the policy.
-- Blocks creating of new passwords for apps and service principals after 2019-10-19 at 10:37 AM UTC time.
+- Blocks creating of new passwords for applications and service principals created on or after 2019-10-19 at 10:37 AM UTC time.
+- Enforces lifetime on passwords secrets,  symmetric and asymmetric key credentials for applications crated on or after 2014-10-19 at 10:37 AM UTC time.
```
Comment (dougsikora) [inline api-reference/beta/api/appManagementPolicy-post.md:59]: extra space here

File: api-reference/beta/resources/applicationAuthenticationMethodPolicy.md [hunk 2]
```diff
@@ -47,7 +47,10 @@ The application authentication methods policy API offers the following restricti
...
 | passwordAddition | Restrict password secrets on applications altogether. | Block new passwords on applications created on or after '01/01/2019'.                        |
-| passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to a maximum of 30 days for on applications created after '01/01/2019'. |
+| passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to a maximum of 30 days for on applications created after '01/01/2015'. |
+| symmetricKeyAddition | Restrict symmetric keys on applications altogether. | Block new symmetric keys on applications created on or after '01/01/2019'. |
+| symmetricKeyLifetime | Enforce a max lifetime range for a symmetric key.   | Restrict all new symmetric keys to a maximum of 30 days for on applications created after '01/01/2019'. |
+| asymmetricKeyLifetime | Enforce a max lifetime range for a asymmetric key (certificate).   | Restrict all new asymmetric key secrets to a maximum of 30 days for on applications created after '01/01/2019'. |
```
Comment (dougsikora) [inline api-reference/beta/resources/applicationAuthenticationMethodPolicy.md:53]:
```suggestion
| asymmetricKeyLifetime | Enforce a max lifetime range for an asymmetric key (certificate).   | Restrict all new asymmetric key secrets to a maximum of 30 days for on applications created after '01/01/2019'. |
```

File: api-reference/beta/resources/applicationAuthenticationMethodPolicy.md [hunk 3]
```diff
@@ -47,7 +47,10 @@ The application authentication methods policy API offers the following restricti
...
 | :--------------- | :---------------------------------------------------- | :------------------------------------------------------------------------------------------- |
 | passwordAddition | Restrict password secrets on applications altogether. | Block new passwords on applications created on or after '01/01/2019'.                        |
-| passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to a maximum of 30 days for on applications created after '01/01/2019'. |
+| passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to a maximum of 30 days for on applications created after '01/01/2015'. |
+| symmetricKeyAddition | Restrict symmetric keys on applications altogether. | Block new symmetric keys on applications created on or after '01/01/2019'. |
+| symmetricKeyLifetime | Enforce a max lifetime range for a symmetric key.   | Restrict all new symmetric keys to a maximum of 30 days for on applications created after '01/01/2019'. |
```
Comment (dougsikora) [inline api-reference/beta/resources/applicationAuthenticationMethodPolicy.md:52]:
```suggestion
| symmetricKeyLifetime | Enforce a max lifetime range for a symmetric key.   | Restrict all new symmetric keys to a maximum of 30 days for applications created after '01/01/2019'. |
```

File: api-reference/beta/resources/applicationAuthenticationMethodPolicy.md [hunk 4]
```diff
@@ -47,7 +47,10 @@ The application authentication methods policy API offers the following restricti
 | Restriction name      | Description                                           | Examples                                                                                     |
 | :--------------- | :---------------------------------------------------- | :------------------------------------------------------------------------------------------- |
 | passwordAddition | Restrict password secrets on applications altogether. | Block new passwords on applications created on or after '01/01/2019'.                        |
-| passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to a maximum of 30 days for on applications created after '01/01/2019'. |
+| passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to a maximum of 30 days for on applications created after '01/01/2015'. |
```
Comment (dougsikora) [inline api-reference/beta/resources/applicationAuthenticationMethodPolicy.md:50]:
```suggestion
| passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to a maximum of 30 days for applications created after '01/01/2015'. |
```

File: changelog/Microsoft.DirectoryServices.json [hunk 5]
```diff
@@ -1,26 +1,67 @@
...
+          "SubArea": "Directory management"
+          }, {
+            "ChangeList": [{
         "Id": "aa34ca74-c3b4-4981-ba1c-809256deacb0",
         "ApiChange": "Method",
         "ChangedApiName": "Delete",
         "ChangeType": "Addition",
         "Description": "Added the [delete](https://docs.microsoft.com/en-us/graph/api/accesspackageassignmentrequest-delete?view=graph-rest-beta) operation to the [accessPackageAssignmentRequest](https://docs.microsoft.com/en-us/graph/api/resources/accesspackageassignmentrequest?view=graph-rest-beta) resource.",
         "Target": "Delete"
-      }
```
Comment (dougsikora) [inline changelog/Microsoft.DirectoryServices.json:12]: There are a large number of diffs in this file unrelated to the additional change log items added. Can you correct is so that only the item being added shows up as a diff?

Comment (madansr7) [reply changelog/Microsoft.DirectoryServices.json:12]: I think that's vscode trying to format the json on save. Long list of changes is showing due to JSON formatter.

Comment (dougsikora) [reply changelog/Microsoft.DirectoryServices.json:12]: Yea I understand. But, I think if the diff is not clean it will cause these issues:
1. Merge conflicts for you and others
2. Make the code history hard to follow for changes to this file, especially given the length

File: changelog/Microsoft.DirectoryServices.json [hunk 6]
```diff
@@ -1,61 +1,104 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [symmetricKeyAddition](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "appCredentialRestrictionType"
+        },
+        {
+          "Id": "081f222d-0ccd-42bd-aff6-3455f608f9d5",
+          "ApiChange": "Member",
+          "ChangedApiName": "symmetricKeyLifetime",
+          "ChangeType": "Addition",
+          "Description": "Added the [symmetricKeyLifetime](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:16]:
```suggestion
          "Description": "Added the `symmetricKeyLifetime` member to the **appCredentialRestrictionType** enumeration.",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 7]
```diff
@@ -1,61 +1,104 @@
...
+          "Target": "appCredentialRestrictionType"
+        },
+        {
+          "Id": "081f222d-0ccd-42bd-aff6-3455f608f9d5",
+          "ApiChange": "EnumType",
+          "ChangedApiName": "appKeyCredentialRestrictionType",
+          "ChangeType": "Addition",
+          "Description": "Added the [appKeyCredentialRestrictionType](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "appKeyCredentialRestrictionType"
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:25]:
```suggestion
          "Id": "081f222d-0ccd-42bd-aff6-3455f608f9d5",
          "ApiChange": "Enum type",
          "ChangedApiName": "appKeyCredentialRestrictionType",
          "ChangeType": "Addition",
          "Description": "Added the **appKeyCredentialRestrictionType** enumeration.",
          "Target": "appKeyCredentialRestrictionType"
```

File: changelog/Microsoft.DirectoryServices.json [hunk 8]
```diff
@@ -1,61 +1,104 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [keyCredentials](https://docs.microsoft.com/en-us/graph/api/resources/appManagementConfiguration?view=graph-rest-beta) resource",
+          "Target": "appManagementConfiguration"
+        },
+        {
+          "Id": "081f222d-0ccd-42bd-aff6-3455f608f9d5",
+          "ApiChange": "Resource",
+          "ChangedApiName": "keyCredentialConfiguration",
+          "ChangeType": "Addition",
+          "Description": "Added the [keyCredentialConfiguration](https://docs.microsoft.com/en-us/graph/api/resources/keyCredentialConfiguration?view=graph-rest-beta) resource",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:40]:
```suggestion
          "Description": "Added the [keyCredentialConfiguration](https://docs.microsoft.com/en-us/graph/api/resources/keycredentialconfiguration?view=graph-rest-beta) resource.",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 9]
```diff
@@ -1,61 +1,104 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [appKeyCredentialRestrictionType](https://docs.microsoft.com/en-us/?view=graph-rest-beta) resource",
+          "Target": "appKeyCredentialRestrictionType"
+        },
+        {
+          "Id": "081f222d-0ccd-42bd-aff6-3455f608f9d5",
+          "ApiChange": "Property",
+          "ChangedApiName": "keyCredentials",
+          "ChangeType": "Addition",
+          "Description": "Added the [keyCredentials](https://docs.microsoft.com/en-us/graph/api/resources/appManagementConfiguration?view=graph-rest-beta) resource",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:32]:
```suggestion
          "Description": "Added the **keyCredentials** property to the [appManagementConfiguration](https://docs.microsoft.com/en-us/graph/api/resources/appmanagementconfiguration?view=graph-rest-beta) resource.",
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
- Don't forget to document **appKeyCredentialRestrictionType** in Enums.md.
- I'm curious about the changelog entries. I hope these weren't generated by the stub generator/Graph Studio? If they were, kindly confirm so that we can flag this as a bug.
- Recommending that you retry the changelog from a fresh version.

File: api-reference/beta/resources/keyCredentialConfiguration.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+---
+title: "keyCredentialConfiguration resource type"
+description: "Key credential configuration complex type to configure key credential restriction, maxLifetime, and enforcement date"
+author: "madansr7"
+localization_priority: Normal
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/keyCredentialConfiguration.md:5]:
```suggestion
ms.localizationpriority: medium
```
